Documentation/contributing/development/introducing_new_crds.rst

```diff
@@ -0,0 +1,400 @@
+
+   // +kubebuilder:storageversion
+
+You can find CRD generation ``marks`` documentation
+`here <https://book.kubebuilder.io/reference/markers/crd.html>`__.
+
+Marks are also used to generate json-schema validation. You can define
+validation criteria such as “format=cidr” and “required” via validation
+``marks`` in your struct’s comments.
+
+An example:
+
+.. code:: go
+
+   type CiliumBGPPeeringConfiguration struct {
+       // PeerAddress is the IP address of the peer.
+       // This must be in CIDR notation and use a /32 to express
+       // a single host.
+       //
+       // +kubebuilder:validation:Required
+       // +kubebuilder:validation:Format=cidr
+       PeerAddress string `json:"peerAddress"`
+
+You can find CRD validation ``marks`` documentation
+`here <https://book.kubebuilder.io/reference/markers/crd-validation.html>`__.
+
+Defining CRDs
+-------------
+
+With a quick introduction to marks we can start defining CRDs in the
+following default paths:
+
+Paths
+
+::
+
+   pkg/k8s/apis/cilium.io/v2/
+   pkg/k8s/apis/cilium.io/v2alpha1/
+
+The portion of the directory after ``apis/`` makes up the CRD’s
+``Group`` and ``Version``. See
+`KubeBuilder-GVK <https://book.kubebuilder.io/cronjob-tutorial/gvks.html>`__
+
+You can begin defining your ``CRD`` structure, making any subtypes you
+like to adequately define your data model and using ``marks`` to control
+the CRD generation process.
+
+Here is an brief example, omitting any further definitions of sub-types
+to express the CRD data model. \`\`
+
+.. code:: go
+
+   // +genclient
+   // +genclient:nonNamespaced
+   // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+   // +kubebuilder:resource:categories={cilium,ciliumbgp},singular="ciliumbgppeeringpolicy",path="ciliumbgppeeringpolicies",scope="Cluster",shortName={bgpp}
+   // +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+   // +kubebuilder:storageversion
+
+   // CiliumBGPPeeringPolicy is a Kubernetes third-party resource for instructing
+   // Cilium's BGP control plane to create peers.
+   type CiliumBGPPeeringPolicy struct {
+       // +k8s:openapi-gen=false
+       // +deepequal-gen=false
+       metav1.TypeMeta `json:",inline"`
+       // +k8s:openapi-gen=false
+       // +deepequal-gen=false
+       metav1.ObjectMeta `json:"metadata"`
+
+       // Spec is a human readable description of a BGP peering policy
+       //
+       // +kubebuilder:validation:Required
+       Spec CiliumBGPPeeringPolicySpec `json:"spec,omitempty"`
+   }
+
+Integrating CRDs into Cilium
+----------------------------
+
+Once you’ve coded your CRD data model you can use Cilium’s ``Make``
+infrastructure to generate and integrate your CRD into Cilium.
+
+There are several make targets and a script which revolve around
+generating CRD and associated code gen (client, informers, DeepCopy
+impls, DeepEqual impls, etc…)
```

Comment:
Please enquote `DeepCopy` etc., and expand “gen”, “impls”. See also the complaints from the `Documentation` GitHub workflow.
Nit: no ellipsis after `etc` :).